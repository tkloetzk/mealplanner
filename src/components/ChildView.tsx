// src/components/ChildView.tsx
import { Card } from "@/components/ui/card";
import { Check } from "lucide-react";
import { MealType, CategoryType, Food, MealPlan } from "@/types/food";

interface ChildViewProps {
  selectedMeal: MealType | null;
  foodOptions: Record<CategoryType, Food[]>;
  selections: MealPlan;
  selectedDay: string;
  onFoodSelect: (category: CategoryType, food: Food) => void;
  onMealSelect: (meal: MealType) => void;
}

export function ChildView({
  selectedMeal,
  foodOptions,
  selections,
  selectedDay,
  onFoodSelect,
  onMealSelect,
}: ChildViewProps) {
  const MEAL_TYPES: MealType[] = ["breakfast", "lunch", "dinner", "snack"];

  if (!selectedDay) {
    console.log(selectedDay, new Date());
  }
  if (!selectedMeal) {
    return (
      <div className="text-center py-12">
        <h2 className="text-2xl font-bold mb-6">What are you eating?</h2>
        <div className="grid grid-cols-2 gap-4 max-w-md mx-auto">
          {MEAL_TYPES.map((meal) => (
            <Card
              key={meal}
              className="p-6 text-center cursor-pointer hover:bg-blue-50 transition-colors"
              onClick={() => onMealSelect(meal)}
            >
              <div className="text-xl capitalize">{meal}</div>
            </Card>
          ))}
        </div>
      </div>
    );
  }

  const getCategoryEmoji = (category: CategoryType) => {
    switch (category) {
      case "fruits":
        return "üçé";
      case "vegetables":
        return "ü•ï";
      case "proteins":
        return "ü•ö";
      case "grains":
        return "ü•ñ";
      default:
        return "üçΩÔ∏è";
    }
  };

  // Add multiple border style options for each category
  const getCategoryStyles = (category: CategoryType) => {
    switch (category) {
      case "fruits":
        return "border-l-4 border-red-400 border-r border-t border-b";
      case "vegetables":
        return "border-l-4 border-green-400 border-r border-t border-b";
      case "proteins":
        return "border-l-4 border-blue-400 border-r border-t border-b";
      case "grains":
        return "border-l-4 border-yellow-400 border-r border-t border-b";
      default:
        return "border";
    }
  };

  // Function to get the image source
  const getImageSource = (food: Food) => {
    if (food.imagePath) {
      return `/images/food/${food.imagePath}`;
    }
    if (food.imageUrl) {
      return food.imageUrl;
    }
    return null;
  };

  return (
    <div className="p-4">
      <h2 className="text-2xl font-bold text-center mb-8 capitalize">
        {selectedMeal}
      </h2>{" "}
      {/* Increased spacing between groups */}
      <div className="space-y-10">
        {(Object.entries(foodOptions) as [CategoryType, Food[]][]).map(
          ([category, foods]) => {
            // Filter foods based on the selected meal
            const compatibleFoods = selectedMeal
              ? foods.filter((food) => food.meal?.includes(selectedMeal))
              : foods;

            // Only render the category if it has compatible foods
            if (compatibleFoods.length === 0) {
              return null;
            }
            return (
              <div key={category} className="relative animate-fade-in">
                <div
                  className={`rounded-xl bg-white shadow-sm ${getCategoryStyles(
                    category
                  )}`}
                >
                  <div
                    className={`px-6 py-3 border-b rounded-t-xl ${getCategoryStyles(
                      category
                    )}`}
                  >
                    <div className="flex items-center gap-2">
                      <span className="text-2xl">
                        {getCategoryEmoji(category)}
                      </span>
                      <h3 className="text-xl font-semibold capitalize">
                        Choose your {category}
                      </h3>
                    </div>
                  </div>

                  {/* Food grid with padding */}
                  <div className="p-6">
                    <div className="grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-4">
                      {compatibleFoods.map((food) => {
                        const isSelected =
                          selections?.[selectedDay]?.[selectedMeal]?.[category]
                            ?.name === food.name;
                        const imageSource = getImageSource(food);

                        return (
                          <Card
                            key={food.name}
                            className={`relative cursor-pointer transition-transform hover:scale-105 
                        ${
                          isSelected
                            ? "ring-2 ring-green-500"
                            : "hover:shadow-md"
                        }`}
                            onClick={() => onFoodSelect(category, food)}
                          >
                            <div className="aspect-square relative overflow-hidden rounded-t-lg">
                              {imageSource ? (
                                <img
                                  src={imageSource}
                                  alt={food.name}
                                  className="object-cover w-full h-full"
                                />
                              ) : (
                                <div className="w-full h-full bg-gray-100 flex items-center justify-center">
                                  <span className="text-4xl">
                                    {category === "fruits"
                                      ? "üçé"
                                      : category === "vegetables"
                                      ? "ü•ï"
                                      : category === "proteins"
                                      ? "ü•ö"
                                      : category === "grains"
                                      ? "ü•ñ"
                                      : "üçΩÔ∏è"}
                                  </span>
                                </div>
                              )}
                              {isSelected && (
                                <div className="absolute top-2 right-2 bg-green-500 rounded-full p-2">
                                  <Check className="w-5 h-5 text-white" />
                                </div>
                              )}
                            </div>

                            {/* Food Info */}
                            <div className="p-3 text-center">
                              <h3 className="font-medium text-lg">
                                {food.name}
                              </h3>
                            </div>
                          </Card>
                        );
                      })}
                    </div>
                  </div>
                </div>
              </div>
            );
          }
        )}
      </div>
    </div>
  );
}
