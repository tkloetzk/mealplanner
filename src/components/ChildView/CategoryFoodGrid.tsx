import { Card } from "@/components/ui/card";
import { Check } from "lucide-react";
import { CategoryType, Food, MealType, DayType, MealPlan } from "@/types/food";

// Define type-safe category styles mapping
const CATEGORY_STYLES: Record<CategoryType, string> = {
  fruits: "border-l-4 border-red-400 border-r border-t border-b",
  vegetables: "border-l-4 border-green-400 border-r border-t border-b",
  proteins: "border-l-4 border-blue-400 border-r border-t border-b",
  grains: "border-l-4 border-yellow-400 border-r border-t border-b",
  milk: "border-l-4 border-purple-400 border-r border-t border-b",
} as const;

// Create type-safe emoji mapping
const CATEGORY_EMOJIS: Record<CategoryType, string> = {
  fruits: "🍎",
  vegetables: "🥕",
  proteins: "🥚",
  grains: "🥖",
  milk: "🥛",
} as const;

interface CategoryFoodGridProps {
  category: CategoryType;
  foods: Food[];
  selectedDay: DayType;
  selectedMeal: MealType;
  selections: MealPlan;
  onFoodSelect: (category: CategoryType, food: Food) => void;
}

// Helper function to get image source with proper typing
const getImageSource = (food: Food): string | null => {
  if (food.imagePath) {
    return `/images/food/${food.imagePath}`;
  }
  return food.imageUrl ?? null;
};

// Helper function to check if a food is selected
const isFoodSelected = (
  selections: MealPlan,
  selectedDay: DayType,
  selectedMeal: MealType,
  category: CategoryType,
  foodName: string
): boolean => {
  return selections[selectedDay]?.[selectedMeal]?.[category]?.name === foodName;
};

export function CategoryFoodGrid({
  category,
  foods,
  selectedDay,
  selectedMeal,
  selections,
  onFoodSelect,
}: CategoryFoodGridProps) {
  return (
    <div className="relative animate-fade-in">
      <div
        className={`rounded-xl bg-white shadow-sm ${CATEGORY_STYLES[category]}`}
      >
        {/* Category Header */}
        <div
          className={`px-6 py-3 border-b rounded-t-xl ${CATEGORY_STYLES[category]}`}
        >
          <div className="flex items-center gap-2">
            <span className="text-2xl">{CATEGORY_EMOJIS[category]}</span>
            <h3 className="text-xl font-semibold capitalize">
              Choose your {category}
            </h3>
          </div>
        </div>

        {/* Food Grid */}
        <div className="p-6">
          <div className="grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-4">
            {foods.map((food) => {
              const isSelected = isFoodSelected(
                selections,
                selectedDay,
                selectedMeal,
                category,
                food.name
              );
              const imageSource = getImageSource(food);

              return (
                <Card
                  key={food.name}
                  className={`relative cursor-pointer transition-transform hover:scale-105 ${
                    isSelected ? "ring-2 ring-green-500" : "hover:shadow-md"
                  }`}
                  onClick={() => onFoodSelect(category, food)}
                >
                  <div className="aspect-square relative overflow-hidden rounded-t-lg">
                    {imageSource ? (
                      <img
                        src={imageSource}
                        alt={food.name}
                        className="object-cover w-full h-full"
                      />
                    ) : (
                      <div className="w-full h-full bg-gray-100 flex items-center justify-center">
                        <span className="text-4xl">
                          {CATEGORY_EMOJIS[category]}
                        </span>
                      </div>
                    )}
                    {isSelected && (
                      <div className="absolute top-2 right-2 bg-green-500 rounded-full p-2">
                        <Check className="w-5 h-5 text-white" />
                      </div>
                    )}
                  </div>
                  <div className="p-3 text-center">
                    <h3 className="font-medium text-lg">{food.name}</h3>
                  </div>
                </Card>
              );
            })}
          </div>
        </div>
      </div>
    </div>
  );
}
