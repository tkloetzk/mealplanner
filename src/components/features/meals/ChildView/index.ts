export { ChildView } from "./ChildView";
export { CategoryFoodGrid } from "./CategoryFoodGrid";
