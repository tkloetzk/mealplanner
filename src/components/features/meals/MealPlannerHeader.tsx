// components/features/meals/MealPlannerHeader.tsx
import { KidSelector } from "@/components/common/KidSelector";
import { ViewToggle } from "@/components/common/ViewToggle";
import { Kid } from "@/types/user";

interface MealPlannerHeaderProps {
  kids: Kid[];
  selectedKid: string;
  isChildView: boolean;
  onKidSelect: (kidId: string) => void;
  onViewToggle: (isChild: boolean) => void;
}

export function MealPlannerHeader({
  kids,
  selectedKid,
  isChildView,
  onKidSelect,
  onViewToggle,
}: MealPlannerHeaderProps) {
  return (
    <>
      <div className="flex justify-between items-center mb-6">
        <h1 className="text-3xl font-bold">Meal Planner</h1>
        <ViewToggle isChildView={isChildView} onToggle={onViewToggle} />
      </div>

      <KidSelector
        kids={kids}
        selectedKid={selectedKid}
        onSelect={onKidSelect}
      />
    </>
  );
}
