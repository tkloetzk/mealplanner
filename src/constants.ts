export type MealType = "breakfast" | "lunch" | "dinner" | "snack";
export const MEAL_TYPES = ["breakfast", "lunch", "dinner", "snack"] as const;
