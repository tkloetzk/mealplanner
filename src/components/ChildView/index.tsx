import { MealType, CategoryType, Food, MealPlan, DayType } from "@/types/food";
import { MealSelector } from "./MealSelector";
import { CategoryFoodGrid } from "./CategoryFoodGrid";

interface ChildViewProps {
  selectedMeal: MealType | null;
  foodOptions: Record<CategoryType, Food[]>;
  selections: MealPlan;
  selectedDay: DayType;
  onFoodSelect: (category: CategoryType, food: Food) => void;
  onMealSelect: (meal: MealType) => void;
}

export function ChildView({
  selectedMeal,
  foodOptions,
  selections,
  selectedDay,
  onFoodSelect,
  onMealSelect,
}: ChildViewProps) {
  // Initial meal selection view
  if (!selectedMeal) {
    return <MealSelector onMealSelect={onMealSelect} />;
  }

  // Food selection view
  return (
    <div className="p-4">
      <h2 className="text-2xl font-bold text-center mb-8 capitalize">
        {selectedMeal}
      </h2>
      <div className="space-y-10">
        {(Object.entries(foodOptions) as Array<[CategoryType, Food[]]>).map(
          ([category, foods]) => {
            // Filter foods for current meal type
            const compatibleFoods = foods.filter((food) =>
              food.meal?.includes(selectedMeal)
            );

            if (compatibleFoods.length === 0) return null;

            return (
              <CategoryFoodGrid
                key={category}
                category={category}
                foods={compatibleFoods}
                selectedDay={selectedDay}
                selectedMeal={selectedMeal}
                selections={selections}
                onFoodSelect={onFoodSelect}
              />
            );
          }
        )}
      </div>
    </div>
  );
}
