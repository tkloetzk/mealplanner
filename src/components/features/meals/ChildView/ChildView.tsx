import { MealType, CategoryType, Food, MealPlan, DayType } from "@/types/food";
import { CategoryFoodGrid } from "./CategoryFoodGrid";
import { MealSelector } from "@/components/features/meals/MealPlanner/components/MealSelector";

interface ChildViewProps {
  selectedMeal: MealType | null;
  foodOptions: Record<CategoryType, Food[]>;
  selections: MealPlan;
  selectedDay: DayType;
  onFoodSelect: (category: CategoryType, food: Food) => void;
  onMealSelect: (meal: MealType) => void;
}

export function ChildView({
  selectedMeal,
  foodOptions,
  selections,
  selectedDay,
  onFoodSelect,
  onMealSelect,
}: ChildViewProps) {
  // Initial meal selection view
  if (!selectedMeal) {
    return <MealSelector onMealSelect={onMealSelect} />;
  }

  // Food selection view
  return (
    <div className="p-4">
      <h2 className="text-2xl font-bold text-center mb-8 capitalize">
        {selectedMeal ? String(selectedMeal) : ""}
      </h2>
      <div className="space-y-10">
        {(Object.entries(foodOptions) as Array<[CategoryType, Food[]]>).map(
          ([category, foods]) => {
            // Filter foods for current meal type
            const compatibleFoods = foods.filter((food) =>
              food.meal?.includes(selectedMeal)
            );

            if (compatibleFoods.length === 0) return null;
            const visibleFoods = foods.filter((food) => !food.hiddenFromChild);

            return (
              <CategoryFoodGrid
                key={category}
                // @ts-expect-error Idk what to do
                category={category}
                foods={visibleFoods} // Use filtered foods
                selectedDay={selectedDay}
                selectedMeal={selectedMeal}
                selections={selections}
                onFoodSelect={onFoodSelect}
              />
            );
          }
        )}
      </div>
    </div>
  );
}
