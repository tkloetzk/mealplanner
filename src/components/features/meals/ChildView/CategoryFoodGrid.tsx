// src/components/features/meals/ChildView/CategoryFoodGrid.tsx

import { Card } from "@/components/ui/card";
import { Check } from "lucide-react";
import { CategoryType, Food, MealType, DayType, MealPlan } from "@/types/food";
import { getFoodImageSource } from "@/utils/imageUtils";
import Image from "next/image";
import { CATEGORY_EMOJIS, CATEGORY_STYLES } from "@/constants";

interface CategoryFoodGridProps {
  category: keyof typeof CATEGORY_STYLES;
  foods: Food[];
  selectedDay: DayType;
  selectedMeal: MealType;
  selections: MealPlan;
  onFoodSelect: (category: CategoryType, food: Food) => void;
  isCondimentGrid?: boolean;
}

export function CategoryFoodGrid({
  category,
  foods,
  selectedDay,
  selectedMeal,
  selections,
  onFoodSelect,
  isCondimentGrid = false,
}: CategoryFoodGridProps) {
  const isFoodSelected = (food: Food): boolean => {
    if (isCondimentGrid) {
      // For condiments, check if it exists in the condiments array
      return (
        // @ts-expect-error idk
        selections[selectedDay]?.[selectedMeal]?.condiments?.some(
          // @ts-expect-error idk
          (c: unknown) => c.foodId === food.id
        ) ?? false
      );
    }

    // For other categories, check if it's the selected item
    // @ts-expect-error idk
    return selections[selectedDay]?.[selectedMeal]?.[category]?.id === food.id;
  };

  const getServingInfo = (food: Food) => {
    if (!isCondimentGrid) return null;

    // @ts-expect-error idk
    const condiment = selections[selectedDay]?.[selectedMeal]?.condiments?.find(
      // @ts-expect-error idk
      (c) => c.foodId === food.id
    );

    if (!condiment) return null;

    return `${condiment.servings} ${food.servingSizeUnit}`;
  };

  return (
    <div className="relative animate-fade-in">
      <div
        className={`rounded-xl bg-white shadow-sm ${CATEGORY_STYLES[category]}`}
      >
        <div
          className={`px-6 py-3 border-b rounded-t-xl ${CATEGORY_STYLES[category]}`}
        >
          <div className="flex items-center gap-2">
            {/** @ts-expect-error idk */}
            <span className="text-2xl">{CATEGORY_EMOJIS[category]}</span>
            <h3 className="text-xl font-semibold capitalize">
              {isCondimentGrid ? "Add Toppings" : `Choose your ${category}`}
            </h3>
          </div>
        </div>

        <div className="p-6">
          <div className="grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-4">
            {foods.map((food, index) => {
              const isSelected = isFoodSelected(food);
              const imageSource = getFoodImageSource(food);
              const servingInfo = getServingInfo(food);

              return (
                <Card
                  key={food.id}
                  data-testid={`${category}-${selectedMeal}-${index}`}
                  className={`relative cursor-pointer transition-transform hover:scale-105 ${
                    isSelected ? "ring-2 ring-green-500" : "hover:shadow-md"
                  }`}
                  onClick={() => onFoodSelect(category, food)}
                >
                  <div className="aspect-square relative overflow-hidden rounded-t-lg">
                    {imageSource ? (
                      <Image
                        src={imageSource}
                        alt={food.name}
                        fill
                        style={{ objectFit: "cover" }}
                        className="object-cover w-full h-full"
                      />
                    ) : (
                      <div className="w-full h-full bg-gray-100 flex items-center justify-center">
                        <span className="text-4xl">
                          {/** @ts-expect-error idk */}
                          {CATEGORY_EMOJIS[category]}
                        </span>
                      </div>
                    )}
                    {isSelected && (
                      <div className="absolute top-2 right-2 bg-green-500 rounded-full p-2">
                        <Check className="w-5 h-5 text-white" />
                      </div>
                    )}
                  </div>
                  <div className="p-3 text-center">
                    <h3 className="font-medium text-lg">{food.name}</h3>
                    {servingInfo && (
                      <div className="text-sm text-gray-600 mt-1">
                        {servingInfo}
                      </div>
                    )}
                  </div>
                </Card>
              );
            })}
          </div>
        </div>
      </div>
    </div>
  );
}
