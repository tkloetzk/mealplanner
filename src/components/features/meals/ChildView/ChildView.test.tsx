// src/components/__tests__/ChildView.test.tsx
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import {
  FRUITS,
  MOCK_FOODS,
  PROTEINS,
  SELECTED_DAY,
  VEGETABLES,
} from "@/__mocks__/testConstants";
import { DEFAULT_MEAL_PLAN } from "@/constants/meal-goals";
import { MEAL_TYPES } from "@/constants";
import { ChildView } from "./ChildView";
import { MealType } from "@/types/meals";

describe("ChildView Component", () => {
  const mockOnFoodSelect = jest.fn();
  const mockOnMealSelect = jest.fn();

  const defaultProps = {
    selectedMeal: null as MealType | null,
    foodOptions: MOCK_FOODS,
    selections: DEFAULT_MEAL_PLAN,
    selectedDay: SELECTED_DAY,
    onFoodSelect: mockOnFoodSelect,
    onMealSelect: mockOnMealSelect,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders meal selection view when no meal is selected", () => {
    render(<ChildView {...defaultProps} />);

    // Check for meal selection prompt
    expect(screen.getByText(/What are you eating\?/i)).toBeInTheDocument();

    // Verify meal type buttons are present
    MEAL_TYPES.forEach((meal) => {
      expect(screen.getByText(new RegExp(meal, "i"))).toBeInTheDocument();
    });
  });

  it("selects a meal and shows food categories", () => {
    render(<ChildView {...defaultProps} selectedMeal="breakfast" />);

    // Check for breakfast header
    expect(screen.getByText(/breakfast/i)).toBeInTheDocument();

    // Verify food categories are displayed
    [PROTEINS, FRUITS, VEGETABLES].forEach((category) => {
      expect(
        screen.getByText(new RegExp(`Choose your ${category}`, "i"))
      ).toBeInTheDocument();
    });
  });

  it("allows food selection in a category", () => {
    render(<ChildView {...defaultProps} selectedMeal="breakfast" />);

    // Find and click a food item
    const foodItem = screen.getByText(MOCK_FOODS.fruits[0].name);
    fireEvent.click(foodItem);

    expect(mockOnFoodSelect).toHaveBeenCalledWith(
      "fruits",
      MOCK_FOODS.fruits[0]
    );
  });

  it("shows available condiments based on selected foods", () => {
    const selectionsWithFood = {
      ...DEFAULT_MEAL_PLAN,
      [SELECTED_DAY]: {
        ...DEFAULT_MEAL_PLAN[SELECTED_DAY],
        breakfast: {
          ...DEFAULT_MEAL_PLAN[SELECTED_DAY].breakfast,
          proteins: MOCK_FOODS.proteins[0],
        },
      },
    };

    render(
      <ChildView
        {...defaultProps}
        selectedMeal="breakfast"
        selections={selectionsWithFood}
      />
    );

    // Check if condiments section is displayed when a food is selected
    const condimentsSection = screen.getByText(/Add Toppings/i);
    expect(condimentsSection).toBeInTheDocument();
  });

  it("matches snapshot when no meal is selected", () => {
    const { asFragment } = render(<ChildView {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with meal selected", () => {
    const { asFragment } = render(
      <ChildView {...defaultProps} selectedMeal="breakfast" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
