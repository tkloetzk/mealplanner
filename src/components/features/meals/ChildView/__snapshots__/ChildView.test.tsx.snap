// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildView Component matches snapshot when no meal is selected 1`] = `
<DocumentFragment>
  <div
    class="text-center py-12"
  >
    <h2
      class="text-2xl font-bold mb-6"
    >
      What are you eating?
    </h2>
    <div
      class="grid grid-cols-2 gap-4 max-w-md mx-auto"
    >
      <div
        class="rounded-xl border bg-card text-card-foreground shadow p-6 text-center cursor-pointer hover:bg-blue-50 transition-colors"
      >
        <div
          class="text-xl capitalize"
        >
          breakfast
        </div>
      </div>
      <div
        class="rounded-xl border bg-card text-card-foreground shadow p-6 text-center cursor-pointer hover:bg-blue-50 transition-colors"
      >
        <div
          class="text-xl capitalize"
        >
          lunch
        </div>
      </div>
      <div
        class="rounded-xl border bg-card text-card-foreground shadow p-6 text-center cursor-pointer hover:bg-blue-50 transition-colors"
      >
        <div
          class="text-xl capitalize"
        >
          dinner
        </div>
      </div>
      <div
        class="rounded-xl border bg-card text-card-foreground shadow p-6 text-center cursor-pointer hover:bg-blue-50 transition-colors"
      >
        <div
          class="text-xl capitalize"
        >
          snack
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChildView Component matches snapshot with meal selected 1`] = `
<DocumentFragment>
  <div
    class="p-4"
    data-testid="child-view"
  >
    <h2
      class="text-2xl font-bold text-center mb-8 capitalize"
    >
      breakfast
    </h2>
    <div
      class="space-y-10"
    >
      <div
        class="relative animate-fade-in"
      >
        <div
          class="rounded-xl bg-white shadow-sm border-l-4 border-blue-400 border-r border-t border-b"
        >
          <div
            class="px-6 py-3 border-b rounded-t-xl border-l-4 border-blue-400 border-r border-t border-b"
          >
            <div
              class="flex items-center gap-2"
            >
              <span
                class="text-2xl"
              >
                ü•ö
              </span>
              <h3
                class="text-xl font-semibold capitalize"
              >
                Choose your proteins
              </h3>
            </div>
          </div>
          <div
            class="p-6"
          >
            <div
              class="grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-4"
            >
              <div
                class="rounded-xl border bg-card text-card-foreground shadow relative cursor-pointer transition-transform hover:scale-105 hover:shadow-md"
                data-testid="proteins-breakfast-0"
              >
                <div
                  class="aspect-square relative overflow-hidden rounded-t-lg"
                >
                  <div
                    class="w-full h-full bg-gray-100 flex items-center justify-center"
                  >
                    <span
                      class="text-4xl"
                    >
                      ü•ö
                    </span>
                  </div>
                </div>
                <div
                  class="p-3 text-center"
                >
                  <h3
                    class="font-medium text-lg"
                  >
                    Chicken Breast
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative animate-fade-in"
      >
        <div
          class="rounded-xl bg-white shadow-sm border-l-4 border-red-400 border-r border-t border-b"
        >
          <div
            class="px-6 py-3 border-b rounded-t-xl border-l-4 border-red-400 border-r border-t border-b"
          >
            <div
              class="flex items-center gap-2"
            >
              <span
                class="text-2xl"
              >
                üçé
              </span>
              <h3
                class="text-xl font-semibold capitalize"
              >
                Choose your fruits
              </h3>
            </div>
          </div>
          <div
            class="p-6"
          >
            <div
              class="grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-4"
            >
              <div
                class="rounded-xl border bg-card text-card-foreground shadow relative cursor-pointer transition-transform hover:scale-105 hover:shadow-md"
                data-testid="fruits-breakfast-0"
              >
                <div
                  class="aspect-square relative overflow-hidden rounded-t-lg"
                >
                  <div
                    class="w-full h-full bg-gray-100 flex items-center justify-center"
                  >
                    <span
                      class="text-4xl"
                    >
                      üçé
                    </span>
                  </div>
                </div>
                <div
                  class="p-3 text-center"
                >
                  <h3
                    class="font-medium text-lg"
                  >
                    Apple
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative animate-fade-in"
      >
        <div
          class="rounded-xl bg-white shadow-sm border-l-4 border-green-400 border-r border-t border-b"
        >
          <div
            class="px-6 py-3 border-b rounded-t-xl border-l-4 border-green-400 border-r border-t border-b"
          >
            <div
              class="flex items-center gap-2"
            >
              <span
                class="text-2xl"
              >
                ü•ï
              </span>
              <h3
                class="text-xl font-semibold capitalize"
              >
                Choose your vegetables
              </h3>
            </div>
          </div>
          <div
            class="p-6"
          >
            <div
              class="grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-4"
            >
              <div
                class="rounded-xl border bg-card text-card-foreground shadow relative cursor-pointer transition-transform hover:scale-105 hover:shadow-md"
                data-testid="vegetables-breakfast-0"
              >
                <div
                  class="aspect-square relative overflow-hidden rounded-t-lg"
                >
                  <div
                    class="w-full h-full bg-gray-100 flex items-center justify-center"
                  >
                    <span
                      class="text-4xl"
                    >
                      ü•ï
                    </span>
                  </div>
                </div>
                <div
                  class="p-3 text-center"
                >
                  <h3
                    class="font-medium text-lg"
                  >
                    Broccoli
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="relative animate-fade-in"
      >
        <div
          class="rounded-xl bg-white shadow-sm border-l-4 border-gray-400 border-r border-t border-b"
        >
          <div
            class="px-6 py-3 border-b rounded-t-xl border-l-4 border-gray-400 border-r border-t border-b"
          >
            <div
              class="flex items-center gap-2"
            >
              <span
                class="text-2xl"
              >
                üçΩÔ∏è
              </span>
              <h3
                class="text-xl font-semibold capitalize"
              >
                Choose your other
              </h3>
            </div>
          </div>
          <div
            class="p-6"
          >
            <div
              class="grid grid-cols-2 sm:grid-cols-3 lg:grid-cols-4 gap-4"
            >
              <div
                class="rounded-xl border bg-card text-card-foreground shadow relative cursor-pointer transition-transform hover:scale-105 hover:shadow-md"
                data-testid="other-breakfast-0"
              >
                <div
                  class="aspect-square relative overflow-hidden rounded-t-lg"
                >
                  <div
                    class="w-full h-full bg-gray-100 flex items-center justify-center"
                  >
                    <span
                      class="text-4xl"
                    >
                      üçΩÔ∏è
                    </span>
                  </div>
                </div>
                <div
                  class="p-3 text-center"
                >
                  <h3
                    class="font-medium text-lg"
                  >
                    Visible Snack
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
